pkg/ddc/base/operation.go

```diff
@@ -238,6 +238,19 @@ func (t *TemplateEngine) reconcileComplete(ctx cruntime.ReconcileRequestContext,
 		ctx.Recorder.Eventf(object, v1.EventTypeNormal, common.DataOperationSucceed,
 			"%s %s succeeded", operation.GetOperationType(), object.GetName())
 	}
+
+	// 5. clean up data operation
+	if len(opStatusToUpdate.Conditions) > 0 {
```

Comment:
Can we use more reliable logic to handle the TTL situation? And should this logic be put in the end of the fuction?

Reply:
put the clean up in the end of reconcileComplete due to the following two considerations:
1. If clean up in the beginning of `reconcileComplete`, in the first `reconcileComplete` the job completion time has not  updated, so the clean up actually do in the following `reconcileComplete`, and this need to add requeue in the end. But if the data operation do not set the `ttlSecondsAfterFinished `, the reque is useless.
2. If data operation set `ttlSecondsAfterFinished`, the `CleanUpIfArriveTTL` would calculate the remaining time to clean up, and requeue after this remaining time, and there will be only next one `reconcileComplete` to clean up the data operation. The `reconcileComplete` only execute two times and there is only small overhead.